Dockerfile

```diff
@@ -20,16 +10,10 @@ RUN sha256sum /tmp/binaryen.tar.gz | grep 9f8397a12931df577b244a27c293d7c976bc7e
 # Extract and install wasm-opt
 RUN tar -xf /tmp/binaryen.tar.gz
 RUN mv binaryen-version_*/wasm-opt /usr/local/bin
-RUN rm -rf binaryen-version_*/ /tmp/binaryen.tar.gz
```

Comment:
Nice catch. This removal makes the file unavailable but does not reduce the image size (which was the original intend).

Reply:
Without using the builder pattern, the way to prevent temp files ending up in the final image is to do all of the work in a single `RUN` command

```
RUN tar -xf /tmp/binaryen.tar.gz && \
  mv binaryen-version_*/wasm-opt /usr/local/bin && \
  rm -rf binaryen-version_*/ /tmp/binaryen.tar.gz
```